pkg/controller/dax/cluster/setup.go

```diff
@@ -0,0 +1,233 @@
+package cluster
+
+import (
+	"context"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/dax"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/google/go-cmp/cmp"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/controller"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/dax/v1alpha1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+// SetupCluster adds a controller that reconciles Cluster.
+func SetupCluster(mgr ctrl.Manager, o controller.Options) error {
+	name := managed.ControllerName(svcapitypes.ClusterGroupKind)
+	opts := []option{
+		func(e *external) {
+			e.preObserve = preObserve
+			e.postObserve = postObserve
+			e.preCreate = preCreate
+			e.postCreate = postCreate
+			e.preUpdate = preUpdate
+			e.preDelete = preDelete
+			e.lateInitialize = lateInitialize
+			e.isUpToDate = isUpToDate
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o.ForControllerRuntime()).
+		For(&svcapitypes.Cluster{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.ClusterGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithInitializers(),
+			managed.WithPollInterval(o.PollInterval),
+			managed.WithLogger(o.Logger.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.Cluster, obj *svcsdk.DescribeClustersInput) error {
+	obj.ClusterNames = append(obj.ClusterNames, awsclients.String(meta.GetExternalName(cr)))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.Cluster, resp *svcsdk.DescribeClustersOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+	switch awsclients.StringValue(resp.Clusters[0].Status) {
+	case "available", "modifying":
+		cr.SetConditions(xpv1.Available())
+	case "creating":
+		cr.SetConditions(xpv1.Creating())
+	case "deleting", "stopped", "stopping":
+		cr.SetConditions(xpv1.Unavailable())
+	}
+	return obs, nil
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.Cluster, obj *svcsdk.CreateClusterInput) error {
+	obj.ClusterName = awsclients.String(cr.Name)
```

Comment:
```suggestion
	obj.ClusterName = awsclients.String(meta.GetExternalName(cr))
```

we using here` GetExternalName(cr)` -Users with Compositions needs this ;)

Reply:
think this is a duplicate :)